x/group/client/testutil/tx.go

```diff
@@ -2347,3 +2429,126 @@ func (s *IntegrationTestSuite) createCLIProposal(groupPolicyAddress, proposer, s
 
 	return testutil.WriteToNewTempFile(s.T(), string(bz)).Name()
 }
+
+func (s *IntegrationTestSuite) createAccounts(quantity int) []string {
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx
+	accounts := make([]string, quantity)
+
+	for i := 1; i <= quantity; i++ {
+		memberNumber := uuid.New().String()
+
+		info, _, err := clientCtx.Keyring.NewMnemonic(fmt.Sprintf("member%s", memberNumber), keyring.English, sdk.FullFundraiserPath,
+			keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+		s.Require().NoError(err)
+
+		pk, err := info.GetPubKey()
+		s.Require().NoError(err)
+
+		account := sdk.AccAddress(pk.Address())
+		accounts[i-1] = account.String()
+
+		_, err = banktestutil.MsgSendExec(clientCtx, val.Address, account,
+			sdk.NewCoins(sdk.NewCoin(s.cfg.BondDenom, sdk.NewInt(1000))),
+			s.commonFlags...,
+		)
+		s.Require().NoError(err)
+	}
+	return accounts
+}
+
+func (s *IntegrationTestSuite) createGroupWithMembers(membersWeight, membersAddress []string) string {
+	s.Require().Equal(len(membersWeight), len(membersAddress))
+	val := s.network.Validators[0]
+	clientCtx := val.ClientCtx
+
+	membersValid := s.newValidMembers(membersWeight, membersAddress)
+	membersByte, err := codec.MarshalJSONIndent(codec.NewLegacyAmino(), membersValid)
```

Comment:
@amaurym Hey I did this to marshal it, however I am not sure this is the ideal way, if not could yoi point me out in the right direction.

Reply:
We shouldn't be using legacy amino, but protobuf everywhere.

The codec is inside `s.cfg.Codec`, you can use its MarshalJSON methodd.

Reply:
Oh wait, my bad, we actually do use `std` json for members: https://github.com/cosmos/cosmos-sdk/blob/main/x/group/client/cli/util.go#L13.

So using `json.Marshal` with the []MemberRequest slice is fine here. Sorry for the confusion.

Reply:
Ohh perfect!! thanks man